src/sql/src/plan/statement/dml.rs

```diff
@@ -331,11 +335,39 @@ pub fn plan_explain_plan(
                 scx.require_feature_flag(&vars::ENABLE_EXPLAIN_BROKEN)?;
             }
 
-            crate::plan::Explainee::Query {
+            crate::plan::Explainee::Statement(ExplaineeStatement::Query {
                 raw_plan,
                 row_set_finishing,
                 broken,
-            }
+            })
+        }
+        Explainee::CreateMaterializedView(mut stmt, broken) => {
+            // If we don't force this parameter to Skip planning fails for names
+            // that already exist in the catalog.
+            stmt.if_exists = IfExistsBehavior::Skip;
```

Comment:
> It seems like we should return an error or some output to indicate that the item already exists.

I was optimizing for the following workflow:

1. Write a SQL query `$MV[1]`.
2. Run `EXPLAIN CREATE MATERIALIZED VIEW mv AS $MV[1]`.
3. Refine the definition until the `EXPLAIN` plan looks good.
4. Run `CREATE MATERIALIZED VIEW mv AS $MV[i]` once satisfied with the `EXPLAIN` output of version `i`.
5. Some time passes.
6. We decide that we want to refine the `mv` definition to `$MV[i+1]`.
7. Run `CREATE MATERIALIZED VIEW mv AS $MV[i+1]`.

If I remove this line, we will get the behavior that you expect, but then step (7) will fail because `mv` already exists. Even though the command that is executed in practice will be `CREATE OR REPLACE MATERIALIZED VIEW`, I'm not sure if we should ask for people to think about that detail when they just want to run `EXPLAIN`.

However, if you think this is more consistent I will change this before merging.

Reply:
(Also note that currently EXPLAIN CREATE OR REPLACE MATERIALIZED VIEW is not accepted by the parser, because of the `if self.peek_keywords(&[CREATE, MATERIALIZED, VIEW])`.)

Reply:
In that case I think we should reject `EXPLAIN CREATE MATERIALIZED VIEW IF NOT EXISTS` in the parser as well.

Reply:
My worry is the following:

1. Someone runs `EXPLAIN CREATE MATERIALIZED VIEW IF NOT EXISTS mv AS $MV[i]`.
2. They are happy with the plan.
3. They run `CREATE MATERIALIZED VIEW IF NOT EXISTS mv AS $MV[i]` expecting the plan they just saw to be installed.
4. The materialized view already existed so nothing happens.

Reply:
> In that case I think we should reject `EXPLAIN CREATE MATERIALIZED VIEW IF NOT EXISTS` in the parser as well, if it's not already rejected.

I'm now even more confused because I just added support for `CREATE OR REPLACE MATERIALIZED VIEW`.

My point was that integrity checks around existing names are only relevant if you want to execute the actual statement. If we just want to explain the statement I think we can be more lenient and bypass some of these checks in order to get into a state where we can execute the optimizer pipeline and re-plan the statement even if running the statement might then fail because some constraint is violated.

@jkosh44 it seems that you see possible sources of confusion with that behavior, can you elaborate?

Reply:
I think my point is that it would be better to reject syntax instead of ignore it, because it will lead to less confusion.

Reply:
> 4. The materialized view already existed so nothing happens

I see, so the notice that we would emit in this case is not sufficient. Then how about:

1. I will use the `parse_create_materialized_view`, accepting the full syntax. 
2. If the `if_exists` is set to `IfExistsBehavior::Skip` at the syntax level, I will emit a parser error indicating that the syntax is not supported because `EXPLAIN` always assumes that the view does not exist.
3. If the `if_exists` is set to anything else, I will override it to `IfExistsBehavior::Skip` in the `IfExistsBehavior::Skip` function.

Reply:
What could happen with the proposal above is

1. A user runs `EXPLAIN CREATE MATERIALIZED VIEW mv AS $MV[i]`.
2. They are happy with the plan.
3. They run `CREATE MATERIALIZED VIEW mv AS $MV[i]` expecting the plan they just saw to be installed.
4. The above statement is rejected because the plan already exists, so the user needs to change the syntax to `EXPLAIN OR REPLACE` and hope that the plan has no dependencies.

I think that this behavior is good because it would allow me to write (internal) console tooling that:

1. gets the `SHOW CREATE` output for existing materialized views,
2. constructs an `EXPLAIN` based on the above,
3. compares the `EXPLAIN` output of the above with the `EXPLAIN MATERIALIZED VIEW` to explore how a plan would change (for example if I were to create an additional index).

Reply:
I'll go ahead with the proposal [from this comment](https://github.com/MaterializeInc/materialize/pull/21973#discussion_r1338683325), then we don't have to rely on the notice.